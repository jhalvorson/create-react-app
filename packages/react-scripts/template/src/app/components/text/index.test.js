import React from 'react';
import { shallow } from 'enzyme';
import Text from './';

/**
 * This is an example smoke test, it just renders the component
 */
it('renders without crashing', () => {
  shallow(<Text />);
});

/**
 * Jest has snapshotting built in ðŸ”¥
 */
it('should render some text', () => {
  const wrapper = shallow(
    <Text>Hello Jest!</Text>,
  );
  expect(wrapper).toMatchSnapshot();
});

/**
 * Testing props is pretty straight forward to
 */
it('renders the children passed in', () => {
  const wrapper = shallow(<Text>signal4Lyf</Text>);
  expect(wrapper.prop('children')).toEqual('signal4Lyf');
});
