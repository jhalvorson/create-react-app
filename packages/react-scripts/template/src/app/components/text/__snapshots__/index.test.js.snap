// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render some text 1`] = `
<p
  className="sig-paragraph"
>
  Hello Jest!
</p>
`;
